airflow/utils/task_group.py

```diff
@@ -207,13 +207,23 @@ def __iter__(self):
             else:
                 yield child
 
-    def add(self, task: DAGNode) -> None:
+    def add(self, task: DAGNode) -> DAGNode:
         """Add a task to this TaskGroup.
 
         :meta private:
         """
         from airflow.models.abstractoperator import AbstractOperator
-
+        from airflow.models.xcom_arg import PlainXComArg
+
+        if TaskGroupContext.active:
+            if isinstance(task, PlainXComArg):
+                task = task.operator
```

Comment:
I feel this unpacking code should be the responsibility of `add_to_taskgroup`, it should pass the underlying task in, not the XComArg wrapper.